### PR TITLE
Adds the CPC (Clone Personnel Carrier) module

### PR DESCRIPTION
**Pull request by Lumipharon**
## About The Pull Request
Adds the PERFECTLY BALANCED, WELL THOUGHT OUT cloner module to the APC.
image

You chug around, pumping out cheap clone troopers who can gear up with the noob vendor (or extra gear if you load it in).
Not pictured is the APC gibber, which produces additional biomass, so you can recycle bodies for YET MORE clone soldiers.
## Why It's Good For The Game
Marine spawner.
Balance perfection.
## Changelog
:cl:
add: Added a cloner module for the APC
/:cl:

### CODE REVIEW

**Comment by Tonklaatech** [comment]
The weapon to surpass Metal Gear.

**Comment by FosterGrime-Heymynameslouie** [comment]
spedemrhg instantly

**Comment by 74delta** [comment]
Holy fuck please

**Comment by SimplySmiley123** [comment]
So real.

**Comment by iforgotmeword** [comment]
IMG_0967

same energy

**Comment by ilikesaying** [comment]
robot version when

**Comment by Admiral-Gigaton** [comment]
can you (if its not already implemented in this PR) make it so corpses can be loaded into the gibber from the outside, and/or gib xeno corpses for biomass (no one sends up minion corpses, so it would work out)

**Comment by Wisemonster** [comment]
> can you, if its not already implemented in this PR, make it so corpses can be loaded into the gibber from the outside, and/or gib xeno corpses for biomass (no one sends up minion corpses, so it would work out)
> 
> its confusing and hard to load corpses n crates into an APC, and not many people know how to, so this would help. or you could make the info about loading things in more available to people.
> 
> otherwise, love this PR, hope it doesnt get shot down, and are there any other cons to doing this, besides getting shitty gear?
> 
> also, does this pull from ghost pool, or is it a DNA scanner moment. could i die, and become fully round removed, but my gear dropped onto the APC, where i then clone via ghost spawn, and pick up my shit?
> 
> QUESTIONS!!

To answer some of these, I'm pretty sure these are just the medbay cloners stuck into an apc, so they will work how the medbay cloner currently works

**Comment by Admiral-Gigaton** [comment]
> > can you, if its not already implemented in this PR, make it so corpses can be loaded into the gibber from the outside, and/or gib xeno corpses for biomass (no one sends up minion corpses, so it would work out)
> > its confusing and hard to load corpses n crates into an APC, and not many people know how to, so this would help. or you could make the info about loading things in more available to people.
> > otherwise, love this PR, hope it doesnt get shot down, and are there any other cons to doing this, besides getting shitty gear?
> > also, does this pull from ghost pool, or is it a DNA scanner moment. could i die, and become fully round removed, but my gear dropped onto the APC, where i then clone via ghost spawn, and pick up my shit?
> > QUESTIONS!!
> 
> To answer some of these, I'm pretty sure these are just the medbay cloners stuck into an apc, so they will work how the medbay cloner currently works

I have no idea how the medbay cloner works, i havent been cloned before (besides 1 case but i barely remember)

**Comment by bowlofcereal** [comment]
this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one

i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this

**Comment by lbnesquik** [comment]
> this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one
> 
> i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this

Bowl is right, biomass is expensive to run.

**Comment by Lumipharon** [comment]
>they don't know

**Comment by 00ze-cyclone** [comment]
> > this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one
> > i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this
> 
> Bowl is right, biomass is expensive to run.

Biomass is expensive sure but you can get more biomass, you can’t get more cloners, those are the true bottleneck of the clone army, then it’s just a matter of fundings

**Comment by lbnesquik** [comment]
They did mention a gibber for biomass. It's just, you can't drag xeno bodies inside the APC, can you?

File: code/game/objects/machinery/cloning/cloning.dm [hunk 1]
```diff
@@ -68,7 +68,7 @@ These act as a respawn mechanic growning a body and offering it up to ghosts.
 
 	if(!linked_machine)
 		// Try to find the machine nearby
-		linked_machine = locate() in orange(1, src)
+		linked_machine = locate() in get_step(src, REVERSE_DIR(dir))
```
Comment (TiviPlus) [inline code/game/objects/machinery/cloning/cloning.dm:71]: feels like we should check this in init as well and have a log_mapping fail

**Comment by yyzsong** [comment]
> this better start with a lot of biomass considering it takes like one whole beaker of biomass for one clone and there are 8 of these, so 8 biomass beakers each time it creates one
> 
> i've got a feeling this is going to be a module pick that makes people scream "griefer" because of this

you get like two and a half clones per beaker you'll be fine

**Comment by Looking4bros** [comment]
I'm pretty sure you get 3 clones per 2 beakers, 40u per clone as they give you regular beakers and not large beakers.

**Comment by Wisemonster** [comment]
Might be a good idea to tell where each clone is, cause currently it's not possible to tell which clone is in the apc, and which is in medbay

**Comment by Looking4bros** [comment]
This is 'uge for Vali, instead of waiting 5-20 minutes getting surgery for necro, or lost limbs, return you body to the grinder, and take a damage penalty for having 0 melee skill for a while. It's Vali, you're not really damaging things anyway.

**Comment by 74delta** [comment]
This is incredibly based. 10/10

**Comment by zee11zee** [comment]
I'm repeating here "Baneling but better"

**Review by Kuro020 — CHANGES_REQUESTED** [review]
image
Really brother

**Comment by TiviPlus** [comment]
fix the clone turbo spam issue pls, this module outclasses the other two by far

**Comment by Lauwma** [comment]
Either it's baneling for marines or a point generator for xenos. I'd say close the PR. if marines aren't doing well with their custom finely tunned loadout what makes you think they will perform as fragile clones with generic loadouts.

**Comment by Looking4bros** [comment]
Perhaps... in species_types/vatborns.dm you could make a third Vatborn APC variant and just make it absolute dogwater, Because this gimmick is very funny, and perhaps would be fine if the clones were just soooo bad. All zeros in skills or a larger damage modifier applied to them without progression to full marine strength.

Also, the gibber can be interacted with as a ghost, and it ejects the beaker.

**Comment by Lumipharon** [comment]
> fix the clone turbo spam issue pls, this module outclasses the other two by far

What do you want me to change? I made it because it was funny, not with serious balance in mind, but I'm open to change whatever.
Less cloners? Longer clone time? less biomass per body?

**Comment by 74delta** [comment]
Turn it into a robot version and make it take mats to make instead of biomass so it's more expensive and less common 🧠

**Comment by chizzy376** [comment]
After 5 rounds of using this, I can definitely say that this is the best module for preventing losing the game due to marine life attrition
Some map spawns have a lot of generous corpses and you can pump out a bunch of clones
Only request is that you should be able to click the console on a finished growth vat to re-alert ghosts that the clone is available for taking (on a cooldown) so you don't have to give hints to ghosts over radio

**Comment by TiviPlus** [comment]
> > fix the clone turbo spam issue pls, this module outclasses the other two by far
> 
> What do you want me to change? I made it because it was funny, not with serious balance in mind, but I'm open to change whatever. Less cloners? Longer clone time? less biomass per body?

iirc main issue is that people scrape up the prespawn bodies to get endless clones

**Comment by Pockets-byte** [comment]
have people put revivable bodies into the gibber? because that sounds like that's something people would do

**Comment by Lumipharon** [comment]
> > > fix the clone turbo spam issue pls, this module outclasses the other two by far
> > 
> > 
> > What do you want me to change? I made it because it was funny, not with serious balance in mind, but I'm open to change whatever. Less cloners? Longer clone time? less biomass per body?
> 
> iirc main issue is that people scrape up the prespawn bodies to get endless clones

Changed so mapspawn bodies can no longer be gibbed for biomass.
Changed the base biomass per meat from 5 to 1, so a lot less biomass over all.

Also fixed some vehicle runtimes while I was at it.

**Comment by ivanmixo** [comment]
Don't think adding marine baneling is a good idea, even if it is funny

**Comment by chizzy376** [comment]
> Changed so mapspawn bodies can no longer be gibbed for biomass

Good change! But I'd like to know,

> Changed the base biomass per meat from 5 to 1, so a lot less biomass over all.

Does this mean that instead of needing to gib 2 DNR marines, you need to gib 10 to make enough biomass?

**Comment by TiviPlus** [comment]
Looks to be okish now?

**Pull request merged**